service/history/workflow/cache/cache.go

```diff
@@ -74,14 +77,16 @@ type (
 
 	CacheImpl struct {
 		cache.Cache
-		shard          shard.Context
 		logger         log.Logger
 		metricsHandler metrics.Handler
 
 		nonUserContextLockTimeout time.Duration
 	}
 
-	NewCacheFn func(shard shard.Context) Cache
+	Key struct {
+		WorkflowKey definition.WorkflowKey
+		ShardUUID   string
```

Comment:
kinda weird that this type and fields need to be exported

Reply:
I have to because there is unit test directly modify this. I can add a TODO to update the unit tests and make those two private.

Reply:
I see. probably we should provide a constructor for the key then. not important.